tests/test_interactions.py

```diff
@@ -0,0 +1,618 @@
+    """Tests for TransE interaction function."""
+
+    cls = pykeen.nn.modules.TransEInteraction
+
+    def _exp_score(self, h, r, t, p, power_norm) -> torch.FloatTensor:
+        assert not power_norm
+        return -(h + r - t).norm(p=p, dim=-1)
+
+
+class TransHTests(TranslationalInteractionTests, unittest.TestCase):
+    """Tests for TransH interaction function."""
+
+    cls = pykeen.nn.modules.TransHInteraction
+
+    def _exp_score(self, h, w_r, d_r, t, p, power_norm) -> torch.FloatTensor:  # noqa: D102
+        assert not power_norm
+        h, w_r, d_r, t = strip_dim(h, w_r, d_r, t)
+        h, t = [x - (x * w_r).sum() * w_r for x in (h, t)]
+        return -(h + d_r - t).norm(p=p)
+
+
+class TransRTests(TranslationalInteractionTests, unittest.TestCase):
+    """Tests for TransR interaction function."""
+
+    cls = pykeen.nn.modules.TransRInteraction
+    shape_kwargs = dict(
+        e=3,
+    )
+
+    def test_manual(self):
+        """Manually test the value of the interaction function."""
+        # Compute Scores
+        h = torch.as_tensor(data=[2, 2], dtype=torch.float32).view(1, 2)
+        r = torch.as_tensor(data=[4, 4], dtype=torch.float32).view(1, 2)
+        m_r = torch.as_tensor(data=[5, 5, 6, 6], dtype=torch.float32).view(1, 2, 2)
+        t = torch.as_tensor(data=[2, 2], dtype=torch.float32).view(1, 2)
+        scores = self.instance.score_hrt(h=h, r=(r, m_r), t=t)
+        first_score = scores[0].item()
+        self.assertAlmostEqual(first_score, -32, delta=1.0e-04)
+
+    def _exp_score(self, h, r, m_r, t, p, power_norm) -> torch.FloatTensor:
+        assert power_norm
+        h, r, m_r, t = strip_dim(h, r, m_r, t)
+        h_bot, t_bot = [clamp_norm(x.unsqueeze(dim=0) @ m_r, p=2, dim=-1, maxnorm=1.) for x in (h, t)]
+        return -((h_bot + r - t_bot) ** p).sum()
+
+
+class SETests(TranslationalInteractionTests, unittest.TestCase):
+    """Tests for SE interaction function."""
+
+    cls = pykeen.nn.modules.StructuredEmbeddingInteraction
+
+    def _exp_score(self, h, t, r_h, r_t, p, power_norm) -> torch.FloatTensor:
+        assert not power_norm
+        # -\|R_h h - R_t t\|
+        h, t, r_h, r_t = strip_dim(h, t, r_h, r_t)
+        h = r_h @ h.unsqueeze(dim=-1)
+        t = r_t @ t.unsqueeze(dim=-1)
+        return -(h - t).norm(p)
+
+
+class UMTests(TranslationalInteractionTests, unittest.TestCase):
+    """Tests for UM interaction function."""
+
+    cls = pykeen.nn.modules.UnstructuredModelInteraction
+
+    def _exp_score(self, h, t, p, power_norm) -> torch.FloatTensor:
+        assert power_norm
+        # -\|h - t\|
+        h, t = strip_dim(h, t)
+        return -(h - t).pow(p).sum()
+
+
+class SimplEInteractionTests(InteractionTests, unittest.TestCase):
+    """Tests for SimplE interaction function."""
+
+    cls = pykeen.nn.modules.SimplEInteraction
+
+    def _exp_score(self, h, r, t, h_inv, r_inv, t_inv, clamp) -> torch.FloatTensor:
```

Comment:
Is this right? I thought SimplE's interaction function was
```
0.5 * (distmult_interaction(h, r, t) + dismult_interaction(t_inv, r_inf, h_inv)
```

The h and t get switched for the inverse one